qiskit/primitives/statevector_estimator.py

```diff
@@ -0,0 +1,161 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Estimator class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable
+from typing import Optional, Union
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit.quantum_info import SparsePauliOp, Statevector
+from qiskit.utils.optionals import HAS_PYDANTIC
+
+from .base import BaseEstimatorV2
+from .containers import (
+    BasePrimitiveOptions,
+    BasePrimitiveOptionsLike,
+    EstimatorPub,
+    EstimatorPubLike,
+    PrimitiveResult,
+    PubResult,
+    make_data_bin,
+)
+from .containers.dataclasses import mutable_dataclass
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+if HAS_PYDANTIC:
+    from pydantic import Field
+    from pydantic.types import PositiveInt
+else:
+    from dataclasses import field as Field
+
+
+@HAS_PYDANTIC.require_in_instance
+@mutable_dataclass
+class ExecutionOptions(BasePrimitiveOptions):
+    """Options for execution."""
+
+    shots: Optional[PositiveInt] = None
+    seed: Optional[Union[int, np.random.Generator]] = None
+
+
+@HAS_PYDANTIC.require_in_instance
+@mutable_dataclass
+class Options(BasePrimitiveOptions):
+    """Options for the primitives.
+
+    Args:
+        execution: Execution time options. See :class:`ExecutionOptions` for all available options.
+    """
+
+    execution: ExecutionOptions = Field(default_factory=ExecutionOptions)
+
+
+@HAS_PYDANTIC.require_in_instance
+class Estimator(BaseEstimatorV2):
```

Comment:
In light of #11524 and #11527, I recommend that this implementation:
 - call super().__init__(0.0) to indicate that the target precision is no error, because this is a full state simulation
 - not introduce any artificial shot noise via rng.normal
 - not have any options (corollary of above two points)

Reply:
This comment was made when the base estimator branch had precision as a constructor argument. I'd argue for similar behaviour wrt to the run method.

Reply:
Now `super().__init__` does not have `precision`. `Estimator` needs to support arbitrary precisions.